src/sql/undo-remove-site-users.pl

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/perl
+#
+# Generates SQL to undelete all CTools users from specified sites
+#
+# undo-remove-site-users.pl: {site-id-file} {output-file}
+#
+#      site-id-file : text file of specified site-ids (each on separate line)
+#      output-file  : SQL file to be executed
+#
+use strict;
+
+my $CPM_ACTION_LOG_SQL = "insert into CPM_ACTION_LOG (ACTION_TIME, SITE_ID, ACTION_TAKEN) VALUES (CURRENT_TIMESTAMP,'SITE-ID','UNDELETE_MEMBERS');\n";
+my $USER_SQL = "INSERT INTO SAKAI_SITE_USER (SITE_ID, USER_ID, PERMISSION) SELECT SITE_ID, USER_ID, PERMISSION FROM ARCHIVE_SAKAI_SITE_USER where site_id = 'SITE-ID';\n";
+my $REALM_SQL = "INSERT INTO SAKAI_REALM_RL_GR (REALM_KEY,  USER_ID ,ROLE_KEY,ACTIVE,PROVIDED) SELECT REALM_KEY,  USER_ID ,ROLE_KEY,ACTIVE,PROVIDED FROM ARCHIVE_SAKAI_REALM_RL_GR where realm_key in (select realm_key from sakai_realm where realm_id like '/site/SITE-ID%');\n";
+## publish site?
+## my $SITE_SQL = "update SAKAI_SITE set published=1 where site_id = 'SITE-ID'; \n";
+my $COMMIT_SQL = "commit;\n";
+
+main:
+{
```

Comment:
for visual clarity you can use print "\n\n...Started...\n\n"; at the begging of the script